packages/react-native/Libraries/promiseRejectionTrackingOptions.js

```diff
@@ -22,6 +24,35 @@ let rejectionTrackingOptions: $NonMaybeType<Parameters<enable>[0]> = {
       // $FlowFixMe[method-unbinding] added when improving typing for this parameters
       message = Error.prototype.toString.call(rejection);
       const error: Error = (rejection: $FlowFixMe);
+
+      // Print pretty unhandled rejections while on DEV
+      if (__DEV__) {
+        const parseErrorStack = require('./Core/Devtools/parseErrorStack');
+        const symbolicateStackTrace = require('./Core/Devtools/symbolicateStackTrace');
+        const LogBox = require('./LogBox/LogBox').default;
+
+        stack = parseErrorStack(error.stack);
+        stack = await symbolicateStackTrace(stack);
+        if (stack) {
+          const warning =
+            `Possible Unhandled Promise Rejection (id: ${id}):\n` +
+            `${message ?? ''}\n` +
+            `at File: ${stack.codeFrame.fileName}, row: ${stack.codeFrame.location.row}, column: ${stack.codeFrame.location.column}`;
+
+          console.log(stack.codeFrame.content);
+
+          LogBox.addLog({
+            level: 'warn',
+            message: {content: warning, substitutions: []},
+            isComponentError: false,
+            codeFrame: stack.codeFrame,
+            stack: error.stack,
```

Comment:
Can't we just pass the symbolicated `stack` variable here instead of the original obfuscated `error.stack`?

Reply:
Logbox already supports doing the symbolication, can we leverage the logic there?

Reply:
Can you provide more context on how to do that? I can see that he already provided it with the `error.stack` but still the correct first frame wasn't shown

Reply:
You may need to use `LogBox.addException` for this, and update `parseLogBoxException` to treat these as non-fatal

Reply:
@ospfranco can you please give this a try?

Reply:
Nope, doesn't seem to be working, didn't change the level yet, but stack parsing doesn't seem to work. I can take a more detailed look a bit later though.

![Oscar Franco Screen 000111](https://github.com/facebook/react-native/assets/1634213/f414accf-f09f-421b-a9a6-ed0664a9412d)